src/ucp/wireup/wireup_cm.c

```diff
@@ -267,40 +267,94 @@ ucp_cm_ep_sa_data_pack(ucp_ep_h ep, ucp_wireup_sockaddr_data_base_t *sa_data,
                            "sa_data version: %u", sa_data_version);
         sa_data->header = UCP_OBJECT_VERSION_V2 <<
                           UCP_SA_DATA_HEADER_VERSION_SHIFT;
-        if (ucp_ep_config(ep)->key.err_mode == UCP_ERR_HANDLING_MODE_PEER) {
+        if (ucp_ep_config_key(ep)->err_mode == UCP_ERR_HANDLING_MODE_PEER) {
             sa_data->header |= UCP_SA_DATA_FLAG_ERR_MODE_PEER;
         }
 
         return sa_data + 1;
     }
 }
 
+static unsigned
+ucp_wireup_cm_address_pack_flags(const ucp_worker_h worker,
+                                 const ucp_wireup_ep_t *cm_wireup_ep,
+                                 unsigned ep_init_flags)
+{
+    unsigned pack_flags = ucp_cm_address_pack_flags(worker);
+
+    if (cm_wireup_ep->flags & UCP_WIREUP_EP_FLAG_SEND_CLIENT_ID) {
+        pack_flags |= UCP_ADDRESS_PACK_FLAG_CLIENT_ID;
+    }
+
+    if (ep_init_flags & UCP_EP_INIT_CREATE_AM_LANE_ONLY) {
+        pack_flags |= UCP_ADDRESS_PACK_FLAG_AM_ONLY;
+    }
+
+    return pack_flags;
+}
+
+static ucs_status_t
+ucp_wireup_cm_priv_space_check(const ucp_worker_h worker,
+                               ucp_rsc_index_t cm_idx, size_t ucp_addr_size,
+                               ucp_object_version_t sa_data_version,
+                               ucs_log_level_t log_level)
+{
+    size_t sa_data_length = ucp_cm_priv_data_length(ucp_addr_size,
+                                                    sa_data_version);
+
+    if (worker->cms[cm_idx].attr.max_conn_priv < sa_data_length) {
+        ucs_log(log_level, "CM %s private data buffer is too small to pack UCP "
+                "endpoint info, cm max_conn_priv %lu, service data version %u, "
+                "size %lu, address length %lu",
+                ucp_context_cm_name(worker->context, cm_idx),
+                worker->cms[cm_idx].attr.max_conn_priv,
+                sa_data_version, sa_data_length - ucp_addr_size, ucp_addr_size);
+        return UCS_ERR_BUFFER_TOO_SMALL;
+    }
+
+    return UCS_OK;
+}
+
+static ucs_status_t
+ucp_wireup_cm_ep_priv_space_is_enough(const ucp_worker_h worker,
```

Comment:
maybe we can remove this function, and ucp_cm_client_uct_connect_progress will call ucp_address_length+ucp_wireup_cm_priv_space_check directly ?

Reply:
Yes. It can do that. I've removed ```ucp_wireup_cm_ep_priv_space_is_enough```.